askup/utils/general.py

```diff
@@ -92,6 +116,69 @@ def get_user_place_in_rank_list(user_id):
     return 0
 
 
+def get_user_profile_rank_list(rank_user_id, viewer_user_id):
+    """
+    Aquire and return a list of rows for the user profile rank list.
+    """
+    first_items = get_user_profile_rank_list_elements(viewer_user_id, args=(rank_user_id,))
+    result_row_datas, user_is_present = compose_user_profile_rank_list_row_data(first_items, rank_user_id)
+
+    if user_is_present:
+        return result_row_datas
+
+    result_row_datas += [(0, None, None, None, None)]  # adding an ellipsis row
+    user_item = get_user_profile_rank_list_elements(rank_user_id, 'where ranked.id = %s', (rank_user_id,))
```

Comment:
As for me it is completely wrong solution for this project and for actually Django. But it is an architectural suggestion only.

Reply:
Do you mean raw sql?

Reply:
@KhaimovMR You pass part or raw SQL query here.

Reply:
Oh yes. It's made in the DRY concerns. I can make two similar functions but with different where expressions in this case.

Reply:
We will leave all as it is now, and return to this part in refactoring in one of the next iterations.